File: java-spring-security/src/main/java/io/pivotal/metric_registrar/examples/spring_security/ExampleController.java
package io.pivotal.metric_registrar.examples.spring_security;

import io.micrometer.core.instrument.MeterRegistry;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.concurrent.atomic.AtomicLong;

@RestController
public class ExampleController {

    private MeterRegistry registry;
    private AtomicLong custom;

    public ExampleController(MeterRegistry registry) {
        this.registry = registry;
        this.custom = new AtomicLong(0L);
    }

    @GetMapping("/high_latency")
    public ResponseEntity<String> highLatency() throws InterruptedException {
        Thread.sleep(2000);
        return new ResponseEntity<>("{}", null, HttpStatus.OK);
    }

    @GetMapping("/custom_metric")
    public ResponseEntity<String> customMetric(@RequestParam(value="inc", defaultValue="") String increment) {
        AtomicLong customGauge = registry.gauge("custom", this.custom);
        if (!"".equals(increment)) {
            customGauge.incrementAndGet();
        } else {
            customGauge.decrementAndGet();
        }
        
        return new ResponseEntity<>("{}", null, HttpStatus.OK);
    }

    @GetMapping("/simple")
    public ResponseEntity<String> simple() {
        registry.counter("simple").increment();
        return new ResponseEntity<>("{}", null, HttpStatus.OK);
    }
}

File: kotlin/src/main/kotlin/io/pivotal/metric_registrar/examples/kotlin/controllers/ExampleController.kt
package io.pivotal.metric_registrar.examples.kotlin.controllers

import io.micrometer.core.instrument.MeterRegistry
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.management.timer.Timer

@RestController
class ExampleController(@Autowired private val registry: MeterRegistry) {
    @RequestMapping("/simple")
    fun simple(): String {
        return "{}"
    }

    @RequestMapping("/high_latency")
    fun highLatency(): String {
        Thread.sleep(2 * Timer.ONE_SECOND)
        return "{}"
    }

    @GetMapping("/custom_metric")
    fun customMetric(): String {
        registry.counter("custom_metric").increment()
        return "{}"
    }
}
